sql/src/main/java/io/crate/operation/reference/doc/lucene/DocIdCollectorExpression.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.reference.doc.lucene;
+
+import io.crate.metadata.doc.DocSysColumns;
+import io.crate.operation.reference.doc.ColumnReferenceExpression;
+
+public class DocIdCollectorExpression extends
+        LuceneCollectorExpression<Long> implements ColumnReferenceExpression {
+
+    public static final String COLUMN_NAME = DocSysColumns.DOCID.name();
+
+    private CollectorContext context;
+    private long docId;
+
+    @Override
+    public void startCollect(CollectorContext context) {
+        super.startCollect(context);
+        this.context = context;
```

Comment:
set fetchid here instead of context

i also would recomment not calling this thing fetch id, since it is actually the id of the reader in the job, which might be usefull for other use cases too. so describing it as what it really is would be better.